components/batch-system/src/router.rs

```diff
@@ -106,72 +91,32 @@ where
     /// A helper function that tries to unify a common access pattern to
     /// mailbox.
     ///
-    /// Generally, when sending a message to a mailbox, cache should be
-    /// check first, if not found, lock should be acquired.
-    ///
     /// Returns None means there is no mailbox inside the normal registry.
     /// Some(None) means there is expected mailbox inside the normal registry
     /// but it returns None after apply the given function. Some(Some) means
-    /// the given function returns Some and cache is updated if it's invalid.
+    /// the given function returns Some.
     #[inline]
     fn check_do<F, R>(&self, addr: u64, mut f: F) -> CheckDoResult<R>
     where
-        F: FnMut(&BasicMailbox<N>) -> Option<R>,
+        F: FnMut(BasicMailbox<N>) -> Option<R>,
```

Comment:
Can we keep `&BasicMailbox<N>`? So that we don't need to clone mailbox for every message.